MassSpectrometry/MsDataFile.cs

```diff
@@ -133,38 +133,37 @@ public static void WindowModeHelper(ref double[] intensities, ref double[] mArra
 
             foreach (Tuple<double, double> range in ranges)
             {
-                List<MzPeak> Temp = new List<MzPeak>();
+                List<MzPeak> handyLittleListofMzPeaksThatIsOnlyNeededTemporarily = new List<MzPeak>();
                 foreach (MzPeak mzIntensityPair in mzIntensites)
                 {
                     if (mzIntensityPair.Mz > range.Item1 && mzIntensityPair.Mz <= range.Item2)
                     {
-                        Temp.Add(mzIntensityPair);
+                        handyLittleListofMzPeaksThatIsOnlyNeededTemporarily.Add(mzIntensityPair);
                     }
                 }
-                if (Temp.Count > 0)
+                if (handyLittleListofMzPeaksThatIsOnlyNeededTemporarily.Count > 0)
                 {
-                    Temp.Sort((x, y) => y.Intensity.CompareTo(x.Intensity)); //sort tuple in place decending by item 2, reverse by changing x and y
-                    double maxIntensity = Temp.Select(pair => pair.Intensity).ToList().Max();
-                    for (int i = 0; i < Math.Min(Temp.Count, filteringParams.NumberOfPeaksToKeepPerWindow.Value); i++)
+                    handyLittleListofMzPeaksThatIsOnlyNeededTemporarily.OrderByDescending(x=>x.Intensity); //sort tuple in place decending by item 2, reverse by changing x and y
+                    double maxIntensity = handyLittleListofMzPeaksThatIsOnlyNeededTemporarily.Max(x=>x.Intensity);
+                    for (int i = 0; i < Math.Min(handyLittleListofMzPeaksThatIsOnlyNeededTemporarily.Count, filteringParams.NumberOfPeaksToKeepPerWindow.Value); i++)
                     {
                         if (WindowMaxNormalizationToValue.HasValue)
                         {
-                            double normalizedIntensity = Temp[i].Intensity / maxIntensity * (double)WindowMaxNormalizationToValue.Value;
-                            mzIntensites_reduced.Add(new MzPeak(Temp[i].Mz, normalizedIntensity));
+                            double normalizedIntensity = handyLittleListofMzPeaksThatIsOnlyNeededTemporarily[i].Intensity / maxIntensity * WindowMaxNormalizationToValue.Value;
+                            mzIntensites_reduced.Add(new MzPeak(handyLittleListofMzPeaksThatIsOnlyNeededTemporarily[i].Mz, normalizedIntensity));
                         }
                         else
                         {
-                            mzIntensites_reduced.Add(Temp[i]);
+                            mzIntensites_reduced.Add(handyLittleListofMzPeaksThatIsOnlyNeededTemporarily[i]);
                         }
                     }
                 }
             }
 
             // convert merged results to array and sort by m/z
+            mzIntensites_reduced.OrderBy(x => x.Mz);
```

Comment:
OrderBy is not a void method, so it needs to be mzIntensities_reduced = mzIntensities_reduced.OrderBy(x=>x.Mz);

Reply:
i left the sort in place